app/src/main/java/de/westnordost/streetcomplete/quests/smoking/AddSmoking.kt

```diff
@@ -0,0 +1,77 @@
+package de.westnordost.streetcomplete.quests.smoking
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.meta.isShopExpressionFragment
+import de.westnordost.streetcomplete.data.meta.updateWithCheckDate
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.filter
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.CITIZEN
+import de.westnordost.streetcomplete.ktx.arrayOfNotNull
+import de.westnordost.streetcomplete.ktx.containsAny
+
+class AddSmoking : OsmFilterQuestType<SmokingAllowed>() {
+
+    private val elementFilterBasic = """
+        (
+            amenity ~ bar|cafe|pub|biergarten|restaurant|food_court|nightclub|stripclub
+            or leisure ~ outdoor_seating
+            or (
+                (amenity ~ fast_food|ice_cream or shop ~ ice_cream|deli|bakery|coffee|tea|wine)
+                and (
+                    (outdoor_seating and outdoor_seating != no)
+                    or (indoor_seating and indoor_seating != no)
+                )
+            )
+        )
+    """
+
+    override val elementFilter = """
+             nodes, ways, relations with
+             ${elementFilterBasic}
+             and takeaway != only
+             and (!smoking or smoking older today -8 years)
+    """
```

Comment:
Unnecessary indentation:

```suggestion
    override val elementFilter = """
         nodes, ways, relations with
         ${elementFilterBasic}
         and takeaway != only
         and (!smoking or smoking older today -8 years)
    """
```